proxy/src/telemetry/metrics/serve.rs

```diff
@@ -27,29 +40,30 @@ impl Serve {
         }
     }
 
-    fn is_gzip(req: &Request) -> bool {
-        if let Some(accept_encodings) = req
-            .headers()
-            .get::<AcceptEncoding>()
-        {
-            return accept_encodings
-                .iter()
-                .any(|&QualityItem { ref item, .. }| item == &Encoding::Gzip)
-        }
-        false
+    fn is_gzip<B>(req: &Request<B>) -> bool {
+        req.headers()
+            .get_all(header::ACCEPT_ENCODING).iter()
+            .any(|value| {
+                value.to_str().ok()
+                    .map(|value| value.contains("gzip"))
```

Comment:
this seems like a functional change that could potentially bite us.  is `value.contains("gzip")` really sufficient/

Reply:
I believe this should be equivalent to what we were doing previously, using `hyper`'s old typed headers API. The intention is to return true if any `Accept-Encoding` header value contains `gzip`, ignoring any quality items.

Reply:
right, but wouldn't this match a value like "bugzipper"?

Reply:
> I suppose that since we depend on the `regex` crate now, I could replace this with a regex that matches "gzip" followed by a valid q-item or nothing.

I'd be pretty surprised if we needed regexes for this -- should be enough to `.split(",").map(|s| s.strip())` -- though I don't see a `strip` implementation in Rust's std -- it wouldn't be all that hard to implement.

Though, given that this is admin code, a regex could be fine. If we wanted to do something like this generically as a library, though (as we've discussed previously), we should try to avoid using `regex` for something so simple.

Reply:
The reason I was going to use a regex is because I realized that we want to accept any quality-item _except_ `q=0`, which explicitly _disallows_ the encoding; I suppose this could be done with string equality but it becomes a little complex.

Reply:
(I realise that the previous code also had incorrect behaviour, as it didn't test that the quality value != 0)

Reply:
@hawkw i think it's fine to use a regex for now. since this was already a bug, do you think it makes sense to address this separately?

Reply:
Perhaps; however, I just finixhed an implementation that doesn't use a regex and corrects the quality-item behaviour, so 🤷‍♀️

Reply:
You're probably looking for `trim`, which is available on `str`.

The typed header in hyper 0.11.x was indeed splitting and parsing completely. The only bug that may have been is not checking that the `q=0`. Do we need a super robust handler for this? Prometheus just sends the one `gzip` value, right?

Reply:
@seanmonstar PR #968 (against this branch, I just split it out to make the diff smaller) should fix this.